cmd/samples/cron/main.go

```diff
@@ -25,29 +22,70 @@ var (
 	logger *zap.Logger
 )
 
+func main() {
+	var err error
+	logger, err = zap.NewDevelopment()
+	if err != nil {
+		panic(err)
+	}
+
+	var mode string
+	var cron string
+	flag.StringVar(&mode, "m", "", "Mode is worker or trigger.")
+	flag.StringVar(&cron, "cron", "* * * * *", "Crontab schedule.")
+	flag.Parse()
+
+	switch mode {
+	case "worker":
+		w := startWorker()
+		// The workers are supposed to be long running process that should not exit.
+		// Use channel to wait for Ctrl+C.
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, os.Interrupt)
+		<-c
+		// Stop worker, close connection, clean up resources.
+		w.Stop()
+	case "trigger":
+		startWorkflow(cron)
+	default:
+		flag.PrintDefaults()
+	}
+}
+
 // This needs to be done as part of a bootstrap step when the process starts.
 // The workers are supposed to be long running.
-func startWorker() {
+func startWorker() worker.Worker {
 	workerOptions := worker.Options{
-		Logger: logger,
+		HostPort: client.DefaultHostPort,
+		Logger:   logger,
 	}
 
-	worker, err := worker.New(DomainName, TaskListName, workerOptions)
+	w, err := worker.New(TaskListName, workerOptions)
 	if err != nil {
 		logger.Fatal("Unable to create worker", zap.Error(err))
 	}
 
-	worker.RegisterWorkflow(SampleCronWorkflow)
-	worker.RegisterActivity(sampleCronActivity)
+	w.RegisterWorkflow(SampleCronWorkflow)
+	w.RegisterActivity(sampleCronActivity)
 
-	err = worker.Start()
+	err = w.Start()
 	if err != nil {
-		logger.Error("Unable to start worker", zap.Error(err))
+		logger.Fatal("Unable to start worker", zap.Error(err))
 	}
+
+	return w
 }
 
 // Start instance of the workflow.
 func startWorkflow(cron string) {
+	c, err := client.NewClient(client.Options{
```

Comment:
Add comment that the client is a heavyweight object and should be created once per process.